CMakeLists.txt

```diff
@@ -177,15 +177,38 @@ file(
 	${PROJECT_SOURCE_DIR}/dScripts/CppScripts.h
 )
 
+# Compiler flags:
+# Disabled deprecated warnings as the MySQL includes have deprecated code in them. 
+# Disabled misleading indentation as DL_LinkedList from RakNet has a weird indent.
+# Disabled no-register
+# Disabled unknown pragmas because Linux doesn't understand Windows pragmas.
+if(UNIX)
+	if(APPLE)
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++17 -O2 -Wuninitialized -D_GLIBCXX_USE_CXX11_ABI=0 -D_GLIBCXX_USE_CXX17_ABI=0 -fPIC")
+	else()
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++17 -O2 -Wuninitialized -D_GLIBCXX_USE_CXX11_ABI=0 -D_GLIBCXX_USE_CXX17_ABI=0 -static-libgcc -fPIC")
```

Comment:
`-std=gnu++17` is redundant as we already set the standard to C++17 at the beginning of this file 
https://github.com/DarkflameUniverse/DarkflameServer/blob/17b8e7045f9a0f297e53a721ac3953e71143d9f7/CMakeLists.txt#L5
We should also do 
```cmake
set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
in order to enforce the C++17 requirement.

Reply:
There are differences between `gnu++17` and `std++17`, there is a requirement for it

Reply:
> There are differences between `gnu++17` and `std++17`, there is a requirement for it

The [CMake docs](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html) seem to imply that it should be doing `gnu++`, but alright.

Reply:
The standard is not required however.

Reply:
> The standard is not required however.

I recall C++17 being required for something related to Mersenne Twister... If not, I know it at least C++11 is required, so maybe we should change the general standard requirement to reflect that.

Reply:
I don't think it really matters.